mlflow/models/utils.py

```diff
@@ -1317,16 +1324,16 @@ def _enforce_object(data: dict[str, Any], obj: Object, required=True):
 
 
 def _enforce_map(data: Any, map_type: Map, required=True):
-    if not required and data is None:
-        return None
+    if (not required or isinstance(map_type.value_type, AnyType)) and (data is None or data == {}):
```

Comment:
So will we allow the following case?
```
required=True
map_type.value_type= AnyType
data is None
```

Reply:
Yes, since AnyType accepts None

Reply:
Ooh I think the meaning of `required` is unclear then and it's counterintuitive if None is allowed for required field. I understand it's difficult to avoid the confusion based on the existing schema, but can we make it explicit that users can pass None to required anytype field?

Reply:
Let me include an example in the docs in a follow-up PR